frontend/src/pages/RunDetails.tsx

```diff
@@ -343,24 +343,37 @@ class RunDetails extends Page<RunDetailsProps, RunDetailsState> {
         WorkflowParser.createRuntimeGraph(workflow) : undefined;
 
       const breadcrumbs: Array<{ displayName: string, href: string }> = [];
-      if (experiment) {
-        breadcrumbs.push(
-          { displayName: 'Experiments', href: RoutePage.EXPERIMENTS },
-          {
-            displayName: experiment.name!,
-            href: RoutePage.EXPERIMENT_DETAILS.replace(':' + RouteParams.experimentId, experiment.id!)
-          });
+      // If this is an archived run, only show Archive in breadcrumbs, otherwise show
+      // the full path, including the experiment if any.
+      if (runMetadata.storage_state === RunStorageState.ARCHIVED) {
+        breadcrumbs.push({ displayName: 'Archive', href: RoutePage.ARCHIVE });
       } else {
-        breadcrumbs.push(
-          { displayName: 'All runs', href: RoutePage.RUNS }
-        );
+        if (experiment) {
+          breadcrumbs.push(
+            { displayName: 'Experiments', href: RoutePage.EXPERIMENTS },
+            {
+              displayName: experiment.name!,
+              href: RoutePage.EXPERIMENT_DETAILS.replace(':' + RouteParams.experimentId, experiment.id!)
+            });
+        } else {
+          breadcrumbs.push(
+            { displayName: 'All runs', href: RoutePage.RUNS }
+          );
+        }
       }
       const pageTitle = <div className={commonCss.flex}>
         {statusToIcon(runMetadata.status as NodePhase, runDetail.run!.created_at)}
         <span style={{ marginLeft: 10 }}>{runMetadata.name!}</span>
       </div>;
 
-      this.props.updateToolbar({ breadcrumbs, pageTitle, pageTitleTooltip: runMetadata.name });
+      const buttons = new Buttons(this.props, this.refresh.bind(this));
```

Comment:
is it okay to create a new bound function here every time the page refreshes?

Reply:
You mean in terms of performance? It'd be slightly better to bind it once on the class, it's not really concerning if this code doesn't get called very frequently. If it's inside the render method, for example, it would be a little more important, but even then, there's no reason to do perf-tweaking for the run details page.